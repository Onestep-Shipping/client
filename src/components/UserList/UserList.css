.booking-instruction-list, .bol-instruction-list {
  height: 100%;
  list-style-type: none;
  background-color: white;
  margin: 0;
  padding: 0px;
  overflow: hidden; 
  overflow-y: scroll;
}

.booking-instruction-list {
  width: 27%;
}

.bol-instruction-list {
  width: 35%;
}

#selected-item {
  background-color:rgb(236, 236, 236);
}

.bol-instruction-item {
  margin: 5px 10px;
  border-radius: 15px;
  padding: 10px 20px;
  line-height: 25px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  user-select: none;
}
 
.regular-checkbox {
  display: none;
}

.regular-checkbox + label {
  background-color: #fafafa;
  border: 1px solid #cacece;
  box-shadow: 0 1px 2px rgba(0,0,0,0.05), inset 0px -15px 10px -12px rgba(0,0,0,0.05);
  padding: 9px;
  border-radius: 3px;
  display: inline-block;
  position: relative;
}

.regular-checkbox + label:active, .regular-checkbox:checked + label:active {
  box-shadow: 0 1px 2px rgba(0,0,0,0.05), inset 0px 1px 3px rgba(0,0,0,0.1);
}

.regular-checkbox:checked + label {
  background-color: #e9ecee;
  border: 1px solid #adb8c0;
  box-shadow: 0 1px 2px rgba(0,0,0,0.05), inset 0px -15px 10px -12px rgba(0,0,0,0.05), inset 15px 10px -12px rgba(255,255,255,0.1);
  color: #99a1a7;
}

.regular-checkbox:checked + label:after {
  content: '\2714';
  font-size: 18px;
  position: absolute;
  top: -2px;
  left: 2px;
  color: #202020;
}

#profile-pending {
  filter: invert(87%) sepia(66%) saturate(563%) hue-rotate(359deg) brightness(106%) contrast(106%);
}

#profile-completed {
  filter: invert(85%) sepia(39%) saturate(398%) hue-rotate(69deg) brightness(103%) contrast(104%);
}

.profile-image {
  width: 60px;
  border: none;
  border-radius: 50px;
}

.item-header-container {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
}

.booking-id-text {
  font-size: 16px;
  font-weight: 600;
}

.customer-email-text {
  font-size: 14px;
  color: rgb(42, 42, 42);
}