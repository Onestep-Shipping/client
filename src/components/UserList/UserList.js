import './UserList.css';

import React, { useState } from 'react';

import PropTypes from 'prop-types';
import moment from 'moment';
import profileImg from '../../assets/profile-placeholder.png';

const UserList = props => {
  const [currentIndex, setIndex] = useState(0);
  const { setInd, opt, type } = props;
  return (
    <ul className={type + "-instruction-list"}> 
      {opt.map((shipment, i) => {
        const { status, form } = type === "booking" ? 
          shipment.bookingRequest : shipment.billInstruction;
        
        return (
        <li id={(i === currentIndex ? 'selected-item' : '')}
            className="bol-instruction-item"
            onClick={() => {
              setInd(i);
              setIndex(i);
            }} key={i}>
          <div>
            <input 
              type="checkbox" id="checkbox-1-1" 
              className="regular-checkbox" 
              disabled={true}
              checked={status === "Received"} 
            />
            <label htmlFor="checkbox-1-1"></label>
          </div>
          <img id={"profile" + (status === "Received" ? "-completed" : "-pending")}
            className="profile-image" 
            src={profileImg} alt="" />
          <div className="item-header-container">
            <span className="booking-id-text">{shipment.bookedBy.name}</span>
            <span className="customer-email-text">
              From: {shipment.bookedBy.personInCharge.name}
            </span>
          </div>
          <span>
            {moment(form.createdAt).calendar(null, {
              sameDay : 'HH:mm',
              lastDay : 'ddd',
              lastWeek : 'ddd',
              sameElse : 'ddd'
            })}
          </span>
        </li>)
      }
      )}
    </ul>
  );
}

export default UserList;

UserList.propTypes = {
  ind: PropTypes.number,
  setInd: PropTypes.func,
  opt: PropTypes.array,
  type: PropTypes.string,
};
