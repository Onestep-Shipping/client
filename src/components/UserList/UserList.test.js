import React from 'react';
import UserList from './UserList';
import { shallow } from 'enzyme';

const mockIndexChange = jest.fn();

let wrapper = shallow(
  <UserList 
    setInd={mockIndexChange}
    opt={[]} type="bol" />
);

describe('UserList', () => {
  it('should render the UserList Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
});