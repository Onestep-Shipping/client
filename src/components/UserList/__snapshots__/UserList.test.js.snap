// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList should render the UserList Component correctly 1`] = `
<ul
  className="bol-instruction-list"
/>
`;
